spec/services/group_stage_service_spec.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+RSpec.describe GroupStageService do
+  before do
+    @stage = create(:stage)
+    @teams1 = create_list(:team, 4)
+    @teams2 = create_list(:team, 4)
+    @groups = Hash[1 => @teams1, 2 => @teams2].values
+  end
+  describe 'generate_group_stage method' do
```

Comment:
use `#<method name>` for referring to methods (to be consistent with other specs, see controller specs for examples)